### ISSUE TITLE
Best practice: should the patches in the recipe be ported to the upstream repo?

### ISSUE THREAD

**Issue by connortann** [conda-forge/shap-feedstock#84]
### Comment:

I noticed that several patches were added to the repo in #79.   It looks like there are three patches which are applied conditionally depending on the cuda compiler version:

```
    - 0001-Build-for-all-CUDA-11-arches.patch  # [cuda_compiler_version == "11.8"]
    - 0001-Build-for-all-CUDA-12-arches.patch  # [(cuda_compiler_version or "").startswith("12")]
    # Taken and adapted from https://github.com/shap/shap/pull/3311
    - 0002-remove-old-kepler-architecture-for-cuda-build.patch  # [(cuda_compiler_version or "").startswith("12")]
```

Should any of these patches be applied to the upstream `shap` repo? This seems intuitively like it might be a helpful thing to do, but I wanted to first ask advice from maintainers who are experienced with conda and cuda.

@xhochy, may I ask if you have any recommendations on best practices? Is the upstream repo's set of `gencode` flags suitable?

**Comment by xhochy** [comment]
Yes it would be nice if they could be upstreamed. They would need to autodetect the CUDA version and then add the respective flags. We're building here for quite a variety of CUDA architectures, thus the resulting package gets quite large.

**Comment by connortann** [comment]
Id be happy to try and help with that. Do you know of any examples or guides for auto detecting the CUDA version and adding the relevant flags? I'm quite cautious about this as I don't know a way to test the CUDA stuff on GitHub Actions in the upstream repo

**Comment by xhochy** [comment]
> Do you know of any examples or guides for auto detecting the CUDA version and adding the relevant flags?

No, this is the main reason I did not propose them as upstream PRs.
